### PR TITLE
Implement more of the TodoMVC functionality.

### PR DESCRIPTION
**Pull request by derekdreery**
This patch allows all tasks to be toggled complete/incomplete.

There is a bug where when the filter changes, the 'completed' check gets out of sync with its task. I haven't nailed down the exact cause yet.

### CODE REVIEW

File: crates/xilem_html/web_examples/todomvc/src/state.rs [hunk 1]
```diff
@@ -48,21 +78,42 @@ impl AppState {
...
+            Err(e) => {
+                tracing::error!("couldn't load existing todos: {e}");
+                Default::default()
+            }
+        }
+    }
+
+    /// Save the current state to local_storage
+    pub fn save(&self) {
+        let raw = serde_json::to_string(self).unwrap_throw();
```
Comment (Philipp-M) [inline crates/xilem_html/web_examples/todomvc/src/state.rs:98]: I'm not sure if `serde_json` isn't a little bit too heavy for this. I'm thinking about something like `serde_wasm_bindgen`, but I think we can optimize that later.

**Review by Philipp-M — APPROVED** [review]
LGTM, thanks.

**Comment by Philipp-M** [comment]
Btw. I couldn't trigger the bug (in firefox), do you have a way to consistently reproduce this?

**Comment by derekdreery** [comment]
> Btw. I couldn't trigger the bug (in firefox), do you have a way to consistently reproduce this?

It might have been fixed. I'll ignore it for now and report it again if I find a reliable repro.

**Pull request merged**